ansible/roles/test/files/ptftests/inner_hash_test.py

```diff
@@ -0,0 +1,505 @@
+                .format(src_port, outer_ip_src, outer_ip_dst, ip_src, ip_dst, sport, dport, ip_proto))
+        send_packet(self, src_port, nvgre_pkt)
+
+        masked_exp_pkt = Mask(nvgre_pkt)
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "dst")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "src")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.IPv6, "hlim")
+
+        return verify_packet_any_port(self, masked_exp_pkt, self.exp_port_list)
+
+
+    def check_ip_route_vxlan(self, hash_key, src_port, ip_dst, ip_src, dport, sport, ip_proto):
+        '''
+        @summary: Check IP vxlan based inner packet hashing works
+        @param hash_key: hash_key we are varying in this iteration
+        @param src_port: index of port to use for sending packet to switch
+        @param ip_dst: Inner packet Destination IP address
+        @param ip_src: Inner packet Source IP address
+        @param dport: Inner packet dst port
+        @param sport: Inner packet src port
+        @param ip_proto: Inner packet ip protocol
+        '''
+        if ip_network(unicode(self.outer_dst_ip)).version == 4:
+            (matched_index, received) = self.check_ipv4_route_vxlan(hash_key, src_port, ip_dst, ip_src, dport, sport, ip_proto)
+        else:
+            (matched_index, received) = self.check_ipv6_route_vxlan(hash_key, src_port, ip_dst, ip_src, dport, sport, ip_proto)
+
+        assert received
+
+        logging.info("Received packet at " + str(matched_index))
+        time.sleep(0.02)
+
+        return (matched_index, received)
+
+
+    def check_ipv4_route_vxlan(self, hash_key, src_port, ip_dst, ip_src, dport, sport, ip_proto):
+        '''
+        @summary: Check IPv4 vxlan based inner packet hashing works
+        @param hash_key: hash_key we are varying in this iteration
+        @param src_port: index of port to use for sending packet to switch
+        @param ip_dst: Inner packet Destination IP address
+        @param ip_src: Inner packet Source IP address
+        @param dport: Inner packet dst port
+        @param sport: Inner packet src port
+        @param ip_proto: Inner packet ip protocol
+        '''
+        src_mac = self.dataplane.get_mac(0, src_port)
+
+        outer_ip_src = self.outer_src_ip_interval.get_random_ip() if hash_key == 'outer-tuples' else self.outer_src_ip_interval.get_first_ip()
+        outer_ip_dst = self.outer_dst_ip_interval.get_random_ip() if hash_key == 'outer-tuples' else self.outer_dst_ip_interval.get_first_ip()
+        outer_sport = random.randint(0, 65535) if hash_key == 'outer-tuples' else 1234
+
+        pkt = self.generate_inner_pkt(sport, dport, ip_src, ip_dst, ip_proto)
+        vxlan_pkt = simple_vxlan_packet(
+                    eth_dst=self.router_mac,
+                    eth_src=src_mac,
+                    ip_id=0,
```

Comment:
@anish-n why do we need it?

Reply:
Same comment as above, will take a look and get back